engine-agent/src/main/java/io/camunda/zeebe/process/test/engine/agent/EngineControlImpl.java

```diff
@@ -0,0 +1,106 @@
+package io.camunda.zeebe.process.test.engine.agent;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.process.test.api.InMemoryEngine;
+import io.camunda.zeebe.process.test.engine.EngineFactory;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlGrpc.EngineControlImplBase;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.GetRecordsRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.IncreaseTimeRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.IncreaseTimeResponse;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.RecordResponse;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.ResetEngineRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.ResetEngineResponse;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.StartEngineRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.StartEngineResponse;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.StopEngineRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.StopEngineResponse;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.WaitForIdleStateRequest;
+import io.camunda.zeebe.process.test.engine.protocol.EngineControlOuterClass.WaitForIdleStateResponse;
+import io.camunda.zeebe.protocol.record.Record;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import java.time.Duration;
+
+public class EngineControlImpl extends EngineControlImplBase {
+
+  private InMemoryEngine engine;
+
+  public EngineControlImpl(final InMemoryEngine engine) {
+    this.engine = engine;
+  }
+
+  @Override
+  public void startEngine(
+      final StartEngineRequest request,
+      final StreamObserver<StartEngineResponse> responseObserver) {
+    engine.start();
+
+    final StartEngineResponse response = StartEngineResponse.newBuilder().build();
+    responseObserver.onNext(response);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void stopEngine(
+      final StopEngineRequest request, final StreamObserver<StopEngineResponse> responseObserver) {
+    engine.stop();
+
+    final StopEngineResponse response = StopEngineResponse.newBuilder().build();
+    responseObserver.onNext(response);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void resetEngine(
+      final ResetEngineRequest request,
+      final StreamObserver<ResetEngineResponse> responseObserver) {
+    engine.stop();
```

Comment:
❓ Can stopping the engine fail? If so, what happens?

Reply:
It could fail. If it happens an error will be thrown which will cause in a failed gRPC request. In the end it would result in a failed unit test. I've added logging in case this happens so we can see what went wrong. I've never had it actually happen before.